test/com/cloudogu/ces/cesbuildlib/TrivyTest.groovy

```diff
@@ -0,0 +1,124 @@
+        Docker dockerMock = mock(Docker.class)
+        Docker.Image imageMock = mock(Docker.Image.class)
+        when(dockerMock.image(trivyImage)).thenReturn(imageMock)
+        when(imageMock.mountJenkinsUser()).thenReturn(imageMock)
+        when(imageMock.mountDockerSocket()).thenReturn(imageMock)
+        when(imageMock.inside(matches("-v /test/.trivy/.cache:/root/.cache/"), any())).thenAnswer(new Answer<Integer>() {
+            @Override
+            Integer answer(InvocationOnMock invocation) throws Throwable {
+                // mock "sh trivy" so that it returns the expected status code and check trivy arguments
+                Closure closure = invocation.getArgument(1)
+                scriptMock.expectedShRetValueForScript.put(expectedTrivyCommand, expectedStatusCode)
+                Integer statusCode = closure.call() as Integer
+                assertEquals(expectedStatusCode, statusCode)
+                assertEquals(expectedTrivyCommand, scriptMock.getActualShMapArgs().getLast())
+
+                // emulate trivy call with local trivy installation and check that it has the same behavior
+                Files.createDirectories(trivyDir)
+                Process process = trivyExec.exec(Trivy.DEFAULT_TRIVY_VERSION, trivyArguments, workDir)
+                if(process.waitFor(2, TimeUnit.MINUTES)) {
+                    assertEquals(expectedStatusCode, process.exitValue())
+                } else {
+                    process.destroyForcibly()
+                    fail("terminate trivy due to timeout")
+                }
+
+                return expectedStatusCode
+            }
+        })
+        Trivy trivy = new Trivy(scriptMock, Trivy.DEFAULT_TRIVY_VERSION, dockerMock)
+
+        trivy.scanImage(imageName, severityLevel, strategy)
+
+        return scriptMock
+    }
+
+    void testScanImage_successfulTrivyExecution() {
+        // with hopes that this image will never have CVEs
+        String imageName = "hello-world"
+        String severityLevel = TrivySeverityLevel.CRITICAL
+
+        def scriptMock = doTestScan(imageName, severityLevel, TrivyScanStrategy.UNSTABLE, 0)
+
+        assertEquals(false, scriptMock.getUnstable())
+    }
+
+    void testScanImage_unstableBecauseOfCVEs() {
+        // with hopes that this image will always have CVEs
+        String imageName = "alpine:3.18.7"
+        String severityLevel = TrivySeverityLevel.ALL
+
+        def scriptMock = doTestScan(imageName, severityLevel, TrivyScanStrategy.UNSTABLE, 10)
+
+        assertEquals(true, scriptMock.getUnstable())
+    }
+
+    void testScanImage_failBecauseOfCVEs() {
+        // with hopes that this image will always have CVEs
+        String imageName = "alpine:3.18.7"
+        String severityLevel = TrivySeverityLevel.ALL
+
+        def gotException = false
+        try {
+            doTestScan(imageName, severityLevel, TrivyScanStrategy.FAIL, 10)doTestScan(imageName, severityLevel, TrivyScanStrategy.FAIL, 10)
```

Comment:
this looks like double copy pasta?

Reply:
@alexander-dammeier is this a mistake and the second `doTestScan` should be deleted?

Reply:
Done, see https://github.com/cloudogu/ces-build-lib/commit/d2cb3a7e99fbb0bb3c439f2ab1e9594186b88707